HPXMLtoOpenStudio/resources/hvac.rb

```diff
@@ -882,10 +886,18 @@ def self.get_heating_setpoints(hvac_control, year)
     if hvac_control.weekday_heating_setpoints.nil? || hvac_control.weekend_heating_setpoints.nil?
       # Base heating setpoint
       htg_setpoint = hvac_control.heating_setpoint_temp
+      # Apply thermostat offset due to onoff control
```

Comment:
Seems like this handles only one of a few different setpoint input types? For example, it doesn't address the "# 24-hr weekday/weekend heating setpoint schedules" code below. Probably better to update the setpoints after the `if`/`else` block of code rather than inserting separate logic into the `if` and `else` statements. Like this?

```
htg_weekday_setpoints = htg_weekday_setpoints.map { |i| i.map { |j| j - offset_db / 2.0 } }
htg_weekend_setpoints = htg_weekend_setpoints.map { |i| i.map { |j| j - offset_db / 2.0 } }
```

We also allow 8760 setpoints to be entered in a CSV, which doesn't seem to be handled.

The fact that we're changing setpoints makes me a bit nervous that other code may need to be updated. For example, there might be other models that use the original setpoints (i.e., refer to the `hvac_control` inputs) rather than these modified inputs, and thus don't operate correctly. Perhaps everything fine, but I could imagine problems like this.

Reply:
Good point. So it's probably fine when other code references the hvac_control setpoints. Maybe the concern is when other parts of the code reference the E+ setpoint directly then. For example, the natural ventilation and whole house fan EMS program uses sensors on the thermostat setpoints.

I can't help but feel that E+ should treat the entered setpoints as the _midpoint_ of the deadband, so that a user doesn't have to modify their entered setpoints when providing this deadband temperature. What do you think?

Regarding the 8760 setpoint schedule, I recall that we are already having to modify it (to convert from F to C), so perhaps you can modify the code [here](https://github.com/NREL/OpenStudio-HPXML/blob/master/HPXMLtoOpenStudio/resources/schedules.rb#L1431).

Reply:
"Regarding the 8760 setpoint schedule, I recall that we are already having to modify it (to convert from F to C), so perhaps you can modify the code [here](https://github.com/NREL/OpenStudio-HPXML/blob/master/HPXMLtoOpenStudio/resources/schedules.rb#L1431)." I am not sure if I followed, I modified so that these two lines applied to both cases (after if-else work):

    htg_weekday_setpoints = htg_weekday_setpoints.map { |i| i.map { |j| j - offset_db / 2.0 } }
    htg_weekend_setpoints = htg_weekend_setpoints.map { |i| i.map { |j| j - offset_db / 2.0 } }

Or is there anything else we need to change?

Reply:
The if/else does not apply when using 8760 setpoint schedules, so I assume additional changes are still needed, yes. If you run the `base-schedules-detailed-setpoints.xml`, for example, it should be obvious if additional changes are needed.

Reply:
Got it. Yes the 8760 setpoint schedules are not handled here. I looked at the convert_setpoints method at `schedules.rb`, there's some challenge reading in the hpxml onoff thermostat temperature inputs into that method. It looks like a preprocessing on schedule files we get from users (and storing the information in the schedule class if I understood it correctly) and it looks pretty separate from other measures. Do you think we should introduce a hpxml argument of on/off thermostat temperature into the schedule handler? Or is it better to modify it later in the hvac.rb or HPXMLtoOpenStudio/measure.rb? I personally would think the latter one might be better since we read in and handle the inputs in the same place.

Reply:
@shorowit I took a second look at the preheating/precooling EMS programs and the natural ventilation program. 
1. The preheating/precooling EMS program I felt like it mimics the real HVAC systems, so it's fine to use the deadband setpoint to calculate loads (so instead of meeting a constant setpoint, it will operate based on the on/off thermostat setpoints, starting to report preheating when zone temperature meets lower bound and stopping preheating when zone temperature meets higher bound). 
2. And the natural ventilation program I might need some extra suggestions. So currently we're shifting heating down by half deadband degree F, and shifting cooling up by half deadband degree F. So that means if there's no heating/cooling operations, the average of these two(which is used in natural ventilation program), is not changed, eg. if the heating setpoint is 70F, cooling setpoint is 76F, then the average will be 73F, with deadband, heating setpoint is 69F. cooling setpoint is 77F, average is still 73F. But if there's heating/cooling operation, the setpoint schedule values will be adjusted by deadband (going up or down by the thermostat deadband, eg. if heating setpoint is 71F then the schedule value will be at 70F or 72F based on HVAC), it will make the average off by half of the deadband. Is this still a fair assumption? If not, adjusting it back will need some extra EMS efforts to detect zone temperature and see which value the current schedule is at.